aten/src/ATen/templates/TensorBody.h

```diff
@@ -323,6 +323,10 @@ class CAFFE2_API Tensor {
   /// Returns if a `Tensor` has quantized backend.
   bool is_quantized() const;
 
+  /// Returns if a `Tensor` is a meta tensor.  Meta tensors can
+  /// also have other designations.
+  bool is_meta() const;
```

Comment:
Because it isn't all that simple; for example, `is_cpu` has to return true for both CPU, SparseCPU and some other keys. Having the methods here at least lets us change how they are implemented under the hood.